src/scene/morph-instance.js

```diff
@@ -173,22 +177,40 @@ class MorphInstance {
     /**
      * Gets current weight of the specified morph target.
      *
-     * @param {number} index - An index of morph target.
+     * @param {string|number} key - An identifier for the morph target. Either the weight index or the weight name
      * @returns {number} Weight.
      */
-    getWeight(index) {
-        return this._weights[index];
+    getWeight(key) {
+        if (typeof key === 'string') {
```

Comment:
I'd create _getWeightIndex function to avoid duplicating code in set and get. Also, no need to do Number.IsInteger .. Map.get returns undefined when the key does not exist.

Reply:
Done 👍